pkg/types/installconfig.go

```diff
@@ -351,3 +351,87 @@ type BootstrapInPlace struct {
 	// InstallationDisk is the target disk drive for coreos-installer
 	InstallationDisk string `json:"installationDisk"`
 }
+
+// ClusterCreateError defines a custom error type that would help identify where the error occurs
+// during the bootstrap phase of the installation process. This would help identify whether the error
+// comes either from the Kubernetes API failure, the bootstrap failure or a general kubernetes client
+// creation error. In the event of any error, this interface packages the error message and a custom
+// log message that must be neatly presented to the user before termination of the project.
+type ClusterCreateError interface {
+	GetError() error
+	GetLogMessage() string
+}
+
+// APIError defines a custom ClusterCreateError that indicates the API failure of the bootstrap phase.
+// Provides a relevant log message along with the actual error that occurs.
+type APIError struct {
```

Comment:
I don't think we need a custom type for each kind of error. Make `ClusterCreateError` a struct and store the cause detail in a string in the struct. You can still keep the `NewXXXError` functions, which will populate the cause details.